### PR TITLE
Add MSYS2 to be able to build QuickJS

### PR DESCRIPTION
**Pull request by nickva**

### CODE REVIEW

File: bin/install_dependencies.ps1 [hunk 1]
```diff
@@ -34,6 +36,11 @@ if (Test-Path($ChocolateyProfile)) {
 }
 refreshenv
 
+Write-Output " ***** Seting up MSYS ******* "
```
Comment (big-r81) [inline bin/install_dependencies.ps1:39]: Small typo, should be `Setting`

File: bin/shell.ps1 [hunk 2]
```diff
@@ -33,6 +33,7 @@ $env:PATH += ";${elxInstallPath}\bin"
 $env:PATH += ";${vcpkgBase}\bin"
 $env:PATH += ";C:\Program Files (x86)\WiX Toolset v3.11\bin"
 $env:PATH += ";C:\tools\${java11Build}\bin"
+$env:PATH += ";C:\tools\msys64\ucrt64\bin"
```
Comment (big-r81) [inline bin/shell.ps1:36]: Short question, where is this path set?

Comment (nickva) [reply bin/shell.ps1:36]: This is set by the msys2 install after chocolatey installs it. Here we're just making gcc.exe and ld.exe available to the build environment so when we `make` the `couch_quickjs` app we don't have to hard-code `c:\tools\...\gcc.exe` in there and can just use `gcc.exe ...`

File: bin/install_dependencies.ps1 (hunk 1 above)
Comment (nickva) [reply bin/install_dependencies.ps1:39]: Oops, thanks for noticing, Ronny. Will fix it.

Comment (nickva) [reply bin/install_dependencies.ps1]: Fixed it

File: bin/shell.ps1 (hunk 2 above)
Comment (nickva) [reply bin/shell.ps1:36]: This is required for QuickJS https://github.com/apache/couchdb/pull/4627 to build on Windows

**Review by big-r81 — APPROVED** [review]
Great work Nick, thank you!🙏

File: bin/shell.ps1 (hunk 2 above)
Comment (nickva) [reply bin/shell.ps1:36]: I had tired to uninstalled and clean all the msys2 bits, then re-install and it still went to `c:\tools\msys64` so at least on my laptop seems to end up in `c:\tools` I can't see where it's configured to do so...

**Pull request merged**